orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/WaitForUpInstancesTask.groovy

```diff
@@ -140,13 +147,29 @@ class WaitForUpInstancesTask extends AbstractWaitingForInstancesTask {
       targetDesiredSize = newTargetDesiredSize
     } else if (stage.context.desiredPercentage != null) {
       Integer percentage = (Integer) stage.context.desiredPercentage
-      targetDesiredSize = getDesiredInstanceCount(capacity, percentage)
-      splainer.add("setting targetDesiredSize=${targetDesiredSize} based on desiredPercentage=${percentage}% of capacity=${capacity}")
+      targetDesiredSize = getDesiredInstanceCount(currentCapacity, percentage)
+      splainer.add("setting targetDesiredSize=${targetDesiredSize} based on desiredPercentage=${percentage}% of capacity=${currentCapacity}")
     }
 
     return targetDesiredSize
   }
 
+  // If either the configured capacity or current serverGroup has autoscaling diasbled, calculate
+  // targetDesired from the configured capacity. This relaxes the need for clouddriver onDemand
+  // cache updates while resizing serverGroups.
+  static boolean useConfiguredCapacity(Stage stage, Map<String, Integer> current) {
+    Map<String, Integer> configuredCapacity = stage.context.getOrDefault("capacity", [:]) as Map<String, Integer>
+
+    if (!configuredCapacity.containsKey("desired")) {
+      return false
+    }
+
+    return (configuredCapacity.getOrDefault("min", 0) == configuredCapacity.getOrDefault("max", 100) &&
```

Comment:
I was worried about `get("desired") == get("max")` evaluating as true because `null == null`. There's currently no check that `current` contains any values, though line 163 already makes this unnecessary for `configuredCapacity` assuming it can't contain keys with null values. I'll refactor to make this simpler.

Reply:
updated, thanks!